authui/src/websocket.ts

```diff
@@ -13,6 +13,7 @@ function refreshPage() {
 }
 
 export class WebSocketController extends Controller {
+  timeDelay: number = -1;
```

Comment:
## Point 1

It's better to name this `backoffIndex`.
The initial value should be 0 (more intuitive than -1).

## Point 2
The setTimeout should be canceled in case the controller disconnects.

## Point 3
The `backoffIndex` should be reset to 0 when connection can be successfully established.

```typescript
reconnectSetTimeoutHandle: number | null = null;

dispose = () => {
  // ...
  // Point 2
  if (this.reconnectSetTimeoutHandle != null) {
    clearTimeout(this.reconnectSetTimeoutHandle);
  }
  this.reconnectSetTimeoutHandle = null;
}

reconnectWebSocket = () => {
  // Point 1
  const index = this.backoffIndex;
  if (this.backoffIndex < 2) {
    this.backoffIndex += 1;
  }
  
  // Point 2
  this.reconnectSetTimeoutHandle = setTimeout(() => {
    this.reconnectSetTimeoutHandle = null;
    this.connectWebSocket();
  }, Math.pow(2, index) * 1000);
};

connectWebSocket = () => {
  // ...
  this.ws.onopen = (e) => {
    // ...
    // Point 3
    this.backoffIndex = 0;
  }
}
```